hazelcast/src/main/java/com/hazelcast/config/ConfigYamlGenerator.java

```diff
@@ -0,0 +1,1266 @@
+        }
+
+        return listenerConfigsAsList;
+    }
+
+    private static List<Map<String, Object>> getEntryListenerConfigsAsList(
+            List<? extends ListenerConfig> listenerConfigs
+    ) {
+        if (listenerConfigs == null || listenerConfigs.isEmpty()) {
+            return null;
+        }
+
+        List<Map<String, Object>> listenerConfigsAsList = new LinkedList<>();
+
+        for (ListenerConfig listenerConfig : listenerConfigs) {
+            Map<String, Object> listenerConfigAsMap = new LinkedHashMap<>();
+
+            addNonNullToMap(listenerConfigAsMap, "class-name",
+                    classNameOrImplClass(listenerConfig.getClassName(), listenerConfig.getImplementation()));
+            addNonNullToMap(listenerConfigAsMap, "include-value",
+                    listenerConfig.isIncludeValue());
+            addNonNullToMap(listenerConfigAsMap, "local",
+                    listenerConfig.isLocal());
+
+            addNonNullToList(listenerConfigsAsList, listenerConfigAsMap);
+        }
+
+        return listenerConfigsAsList;
+    }
+
+    private static Map<String, Object> getPropertiesAsMap(@SuppressWarnings("rawtypes") Map<String, Comparable> properties) {
+        if (properties == null || properties.isEmpty()) {
+            return null;
+        }
+
+        Map<String, Object> propertiesAsMap = new LinkedHashMap<>();
+
+        for (String key : properties.keySet()) {
+            addNonNullToMap(propertiesAsMap, key, properties.get(key));
+        }
+
+        return propertiesAsMap;
+    }
+
+    private static Map<String, Object> getPropertiesAsMap(Properties properties) {
+        if (properties == null || properties.isEmpty()) {
+            return null;
+        }
+
+        Map<String, Object> propertiesAsMap = new LinkedHashMap<>();
+
+        for (Object key : properties.keySet()) {
+            addNonNullToMap(propertiesAsMap, key.toString(), properties.getProperty(key.toString()));
+        }
+
+        return propertiesAsMap;
+    }
+
+    private static Map<String, Object> getStoreConfigAsMap(
+            boolean enabled,
+            String className,
+            String factoryClassName,
+            Properties properties
+    ) {
+        Map<String, Object> storeConfigAsMap = new LinkedHashMap<>();
+
+        addNonNullToMap(storeConfigAsMap, "enabled", enabled);
+        addNonNullToMap(storeConfigAsMap, "class-name", className);
+        addNonNullToMap(storeConfigAsMap, "factory-class-name", factoryClassName);
+        addNonNullToMap(storeConfigAsMap, "properties", getPropertiesAsMap(properties));
+
+        return storeConfigAsMap;
+    }
+
+    private static Map<String, Object> getMapStoreConfigAsMap(MapStoreConfig mapStoreConfig) {
+        if (mapStoreConfig == null) {
+            return null;
+        }
+
+        // Note that order is different from the ConfigXmlGenerator#classNameOrImplClass().
+        // Here implementation takes priority rather than classname.
```

Comment:
what is the reasoning for this change of order?

Reply:
We already discussed this in the offsite but to inform others, it was to follow the changes in the `ConfigXmlGenerator`. It was different in that place hence I did the same in YAML counterpart, and commented the difference.